QA: Migrated Travis and linters to shared repo

tyearke opened the pull request:
## Description
This pull request removes local Travis code and other QA assets and instead uses [a shared repository](https://github.com/ubccr/xdmod-qa) designed to be used by both Open XDMoD and Open XDMoD modules. As part of this change, syntax- and style-checking development dependencies and their config files have been removed as well.

## Motivation and Context
Maintaining Travis code and style rules in each Open XDMoD repository has proven to be a challenge. Changes to these assets in one repository often take time to propagate to other repositories, with the burden of integrating these changes often being put on someone other than the original contributor.

Additionally, providing fixes for Travis issues can also be a difficult process. With the current architecture, fixes can be made in either a new pull request or in the pull request encountering issues, but both approaches have drawbacks. With the former, affected pull requests must rebase or do some other Git trickery to pick up on the Travis updates. With the latter, the pull request is diluted, and the fixes don't help other pull requests that may be affected.

This new approach resolves both issues by ensuring that the latest Travis code and style rules are always pulled in by a Travis build. When a Travis-specific problem occurs, the issue can be resolved in [`xdmod-qa`](https://github.com/ubccr/xdmod-qa) and the fix will be pulled in simply by selecting "Restart Job" for the failed build.

### Caveats
Because the syntax- and style-checking dependencies and their config files are no longer part of each repo, running checks locally requires a different process:

1. Clone the `xdmod-qa` repo.
1. Install dependencies for `xdmod-qa`.
1. Install the style files to a parent directory of your Open XDMoD repos. Run `[xdmod-qa]/style/install.sh [parent-dir]` to do this easily.

This process only needs to be done once, and because the install script uses symlinks, any style updates will become active as soon as the clone of `xdmod-qa` is updated. When new linters and config files are added to `xdmod-qa`, developers simply need to update their clone and re-run the relevant package managers and installation script.

## Tests Performed
I performed the following test runs of different cases on my fork of this repo:

- [Success](https://travis-ci.org/tyearke/xdmod/builds/280929725)
- [Syntax Errors](https://travis-ci.org/tyearke/xdmod/builds/280933080)
- [Style Errors](https://travis-ci.org/tyearke/xdmod/builds/280940126)
- [Unit Test Errors](https://travis-ci.org/tyearke/xdmod/builds/280939662)
- [Build Errors](https://travis-ci.org/tyearke/xdmod/builds/280943908)
- [Install Errors](https://travis-ci.org/tyearke/xdmod/builds/280944376)
- [Post-Install Test Errors](https://travis-ci.org/tyearke/xdmod/builds/280945417)

## Types of changes
- QA improvement

## Checklist:
- [x] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- ~~I have added tests to cover my changes.~~
- [x] All new and existing tests passed.

**EDIT**: Changed the shared repository from being Travis-specific to being for any QA activities.

plessbd commented:
I wonder if we could use a git hook to automate this a bit further.
something similar to https://gist.github.com/sindresorhus/7996717#file-post-merge but force a checkout / update of the qa repo...

plessbd commented:
we should add the CAVEATS to the README.md

tyearke commented:
@chakrabortyr I've added the new ESLint rules to the `xdmod-qa` repo and reran the syntax and style error tests to verify that the changes worked. I also removed the root `package.json` and `package-lock.json` files from this repo altogether since we do not have any remaining npm dependencies in the root directory and we do not publish Open XDMoD on npm.

@plessbd I've added a couple little sections to the README file that point to instructions in `xdmod-qa`.
